src/containers/signup/Signup.js

```diff
@@ -0,0 +1,219 @@
+import React, { Component } from 'react';
+import Axios from 'axios';
+
+
+export default class SignUp extends Component {
+
+  state = {
+    username: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+    errors: {
+      username: {
+        error: false,
+        message: ''
+      },
+      email: {
+        error: false,
+        message: ''
+      },
+      password: {
+        error: false,
+        message: ''
+      },
+      passwordConfirm: {
+        error: false,
+        message: ''
+      },
+    },
+  }
+
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.renderErrors = this.renderErrors.bind(this);
+  }
+
+  handleChange(e) {
+    let node = e.target.name
+    this.setState(
+      {
+        [e.target.name]: e.target.value
+      }
+    )
+    this.setState(prevState => ({
+      ...prevState,
+      errors: {
+        ...prevState.errors,
+        [node]: {
+          ...prevState.errors.node,
+          message: [],
+          error: false
+        }
+      }
+    }));
+  }
+
+  setPasswordState = (message, error) => {
+    this.setState(prevState => ({
+      ...prevState,
+      errors: {
+        ...prevState.errors,
+        passwordConfirm: {
+          ...prevState.errors.passwordConfirm,
+          message: message,
+          error: error
+        }
+      }
+    }));
+  }
+
+  sendRequest = () => {
+    let res = Axios.post(
+      'http://127.0.0.1:8000/api/users/',
+      {
+        user: {
+          username: this.state.username,
+          email: this.state.email,
+          password: this.state.password,
+        },
+      },
+    );
+
+    return res
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    if (this.state.password !== this.state.confirmPassword) {
+      this.setPasswordState(['Passwords do not match'], true);
```

Comment:
This section is buggy, the if statement does not exit the program flow on the event of an unmatching password and confirm password. As such while testing this functionality I discovered that the form submitted despite the passwords not matching. 
As a solution, I discovered that you can solve by using `return   this.setPasswordState(['Passwords do not match'], true);`

Reply:
Thanks for the feedback. I have added a return statement to exit the 'if' statement if passwords do not match